traffic_ops/build/traffic_ops.spec

```diff
@@ -38,7 +38,7 @@ Requires:         openssl-devel, perl, perl-core, perl-DBD-Pg, perl-DBI, perl-Di
 Requires:         libidn-devel, libcurl-devel, libcap
 Requires:         postgresql13 >= 13.2
 Requires:         perl-JSON, perl-libwww-perl, perl-Test-CPAN-Meta, perl-WWW-Curl, perl-TermReadKey, perl-Crypt-ScryptKDF
-Requires:         python(abi)
+Requires:         python(abi), python3
```

Comment:
Since the `python3` package is only required on CentOS 7, instead of adding `python3` to this line, adding this additional line would work (and avoid an additional dependency on Rocky Linux 8 and up):

```spec
{?el7:Requires: python3}
```

Reply:
Well it's always required, I'm guessing EL8+ just includes it? You're really not supposed to use the OS's special Python interpreter in any case. Also it looks like [you can get Python 3.8 through "AppStream"](https://www.server-world.info/en/note?os=CentOS_8&p=python&f=4) - whatever that is? Seems like a `dnf` command, so can we just trust the package manager to handle that?

Reply:
Yeah I just checked and it looks like Rocky 8 doesn't come with Python 2 or 3:
```shellsession
$ docker run -it rockylinux:8
[root@75b60043fd4e /]# python3
bash: python3: command not found
[root@75b60043fd4e /]# python
bash: python: command not found
```
`dnf install python38` works on Rocky 8 but nothing similar works in CentOS7 - although `yum install python3` works fine on both. So it seems like the correct thing to do is require `python3`, but it does seem that the minimum version should still be 3.6 (despite how much I hate that).

Reply:
Rocky 8 comes with Python 3.6 at `/usr/libexec/platform-python3.6` (provided by the `platform-python` package), which the postinstall script can use:

https://github.com/apache/trafficcontrol/blob/ca5955ca3309534e8656430522745f8d2e39fd99/traffic_ops/install/bin/postinstall#L28-L29

So if you want, you could add Python 3 as a dependency in CentOS 7 only like

```spec
{?el7:Requires: python3}
```

(similar to
https://github.com/apache/trafficcontrol/blob/814de6e540048f401e23f1be7a55296dea7efd22/cache-config/testing/docker/trafficserver/trafficserver-9.spec#L47-L48
)

but adding python3 as a Traffic Ops dependency doesn't really hurt anything either way

Reply:
I'd prefer to leave it in, since the official stance of both [Red Hat](https://developers.redhat.com/blog/2018/11/14/python-in-rhel-8-3#why_not_just__python__) and [Rocky Linux](https://forums.rockylinux.org/t/what-difference-platform-python-and-python3/5560) seems to be that you shouldn't use `platform-python` unless you're building something for the OS itself.

Reply:
Sounds good, but in that case `python(abi)` should be removed